utcp.c

```diff
@@ -1131,33 +1442,31 @@ struct timeval utcp_timeout(struct utcp *utcp) {
 			continue;
 		}
 
-		if(timerisset(&c->conn_timeout) && timercmp(&c->conn_timeout, &now, <)) {
-			errno = ETIMEDOUT;
-			c->state = CLOSED;
-			if(c->recv)
-				c->recv(c, NULL, 0);
-			continue;
+		// check connection timeout
+		if(timerisset(&c->conn_timeout)) {
+			 if(timercmp(&c->conn_timeout, &now, <)) {
+				errno = ETIMEDOUT;
+				c->state = CLOSED;
+				if(c->recv)
+					c->recv(c, NULL, 0);
+				continue;
+			}
+			if(timercmp(&c->conn_timeout, &next, <))
```

Comment:
Well, I think we can postpone this rewrite for a later time when we want to take care of the windows port. The current goal is to get ack'ing and retransmit to work.